src/enumerable.cr

```diff
@@ -303,8 +303,16 @@ module Enumerable(T)
     raise ArgumentError.new("size must be positive") if size <= 0
 
     each_slice(size) do |slice|
-      group = slice.size < size ? slice + [filled_up_with] : slice
-      yield group
+      if slice.size < size
+        paddings = Array(T | U).new
+        (size - slice.size).times do
+          paddings << filled_up_with
+        end
+        group = slice + paddings
+        yield group
+      else
+        yield slice
+      end
```

Comment:
Sounds perfect to me!
I copied the logic in https://github.com/crystal-lang/crystal/pull/3128/commits/1c1334fe5307c7032fd65d69a38177b8c771e6c6. 
Thank you!

(Style adjusted to standardize naming for the returned variable 🙇 )